modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/SkinCleanupTest.java

```diff
@@ -257,6 +260,68 @@ public void testToolBarFocus() {
         assertEquals("first item in toolbar must be focused", bar.getItems().get(0), scene.getFocusOwner());
     }
 
+//-------- TabPane
+    @Test
+    public void testChildrenCountAfterSkinIsReplaced() {
+      TabPane tabPane = new TabPane();
+      tabPane.getTabs().addAll(new Tab("0"), new Tab("1"));
+      showControl(tabPane, false);
+      assertEquals(3, tabPane.getChildrenUnmodifiable().size());
+      TabPaneSkin skin1 = (TabPaneSkin)createAlternativeSkin(tabPane);
+      assertEquals(6, tabPane.getChildrenUnmodifiable().size());
+      tabPane.setSkin(skin1);
+      assertEquals(3, tabPane.getChildrenUnmodifiable().size());
+    }
+
```

Comment:
hmm .. looks a bit unusual (could be me, though :)

- why showControl? just install/replace the skin should be enough?
- the outcome of testing an illegal state (after instantiating the skin and before really setting it as the control's skin at which time we have a skin : control relation of 2:1) is unspecified. F.i. there are skins that replace all children in the constructor (which may or may not be a good idea) that would fail a test similar to this.

How about

    TabPane tabPane ..
    installDefaultSkin(tabPane);
    int children = tabPane.getxxChildren().size();
    replaceSkin(tabPane);
    assertEquals(children, tabPane.getxxChildren().size());

If we decide to go without showing, the other tests should be changed as well :)

Reply:
1. `showControl` is not required. It is now changed to `installDefaultSkin` in all tests.
2. Test did verify an illegal state, Intention was to keep a note of that illegal state. I have changed it as you suggested.